File: Empty/java/src/main/java/com/gmail/l2t45s7e9/java/interactor/ContactDetilsModel.java
package com.gmail.l2t45s7e9.java.interactor;

import com.gmail.l2t45s7e9.java.entity.Contact;
import io.reactivex.rxjava3.core.Single;
public class ContactDetilsModel implements ContactDetailsInteractor {

    private ContactDetailsRepository contactDetailsRepository;

    public ContactDetilsModel(ContactDetailsRepository contactDetailsRepository) {
        this.contactDetailsRepository = contactDetailsRepository;
    }

    @Override
    public Single<Contact> getContactDetailsRepo(String id, int color) {
        return contactDetailsRepository.loadDetailsInformation(id, color);
    }
}

File: Empty/java/src/main/java/com/gmail/l2t45s7e9/java/interactor/ContactDetilsModel.kt
package com.gmail.l2t45s7e9.java.interactor

import com.gmail.l2t45s7e9.java.entity.Contact
import kotlinx.coroutines.flow.Flow

class ContactDetilsModel(private val contactDetailsRepository: ContactDetailsRepository) :
    ContactDetailsInteractor {
    override suspend fun getContactDetailsRepo(id: String?, color: Int): Flow<Contact?> {
        return contactDetailsRepository.loadDetailsInformation(id, color)
    }
}